pkg/scheduler/eventhandlers.go

```diff
@@ -246,7 +246,9 @@ func (sched *Scheduler) deletePodFromCache(obj interface{}) {
 		logger.Error(err, "Scheduler cache RemovePod failed", "pod", klog.KObj(pod))
 	}
 
-	sched.SchedulingQueue.MoveAllToActiveOrBackoffQueue(logger, queue.AssignedPodDelete, pod, nil, nil)
+	if assignedPod(pod) {
```

Comment:
The iff possible scenario might be we receive a `DeletedFinalStateUnknown` here, but since this is a stale obj, we should not rely on the object itself. So I think we're right here.